webob/response.py

```diff
@@ -1248,7 +1270,15 @@ def conditional_response_app(self, environ, start_response):
         * Range               (406 Partial Content; only on GET, HEAD)
         """
         req = BaseRequest(environ)
-        headerlist = self._abs_headerlist(environ)
+
+        # Build headerlist, if there is a Location header, make it absolute
+        headerlist = [
```

Comment:
Brought back `_abs_headerlist`, and added an optimization to the default path to only call it when the HTTP Status Code allows for a `Location` header. If someone has a `Location` header on any other status codes they are not within spec, and it won't be made absolute. If this breaks test suites, then they need to get fixed.